sources/bloodhound/src/lib.rs

```diff
@@ -13,7 +13,57 @@ pub fn look_for_string_in_file(path: &str, search_str: &str) -> Result<bool, io:
         .any(|line| line.unwrap_or_default().contains(search_str)))
 }
 
-/// Executes a command and checks if the give `search_str` is in the output.
+/// Reads a file and checks if the given `search_str` is present in its contents.
+pub fn look_for_strings_in_file(path: &str, search_strs: &[&str]) -> Result<bool, io::Error> {
+    let mut matched = 0;
+
+    let reader = BufReader::new(File::open(path)?);
+    for line in reader.lines() {
+        let content = line.unwrap_or_default();
+        for search_str in search_strs {
+            if content.contains(search_str) {
+                matched += 1;
+            }
+        }
+    }
+
+    Ok(search_strs.len() == matched)
+}
+
+/// Check if a given file contains all provided strings, getting a `CheckerResult` of its findings.
+///
+/// If all `strings_to_match` are found in `path`, the `CheckerResult` returned will have a `CheckStatus::PASS`.
+///
+/// If one or more of `strings_to_match` are not found in `path`, then even if some are found the `CheckerResult`
+/// returned will have a `CheckStatus::FAIL` and the `error` field will contain `unable_to_find_error` as its content.
+///
+/// If `path` cannot be read or there is some other error checking the content of the file, returned status will be
+/// `CheckStatus::SKIP` indicating a manual check will need to be performed and the `error` field will contain the
+/// `unable_to_check_error` value.
+#[macro_export]
+macro_rules! check_file_contains {
```

Comment:
I wonder if this is a macro is a bit overkill here, what's the benefit of having this function as a macro instead of just a plain function?

Reply:
I think either a macro or a function could work here. @bcressey had asked for a macro, so maybe he has some opinions on using one vs the other. But I do think it makes the usage nice and concise.